### DISCUSSION TITLE
Feature Request: S3 metadata functions

### DISCUSSION THREAD

**Discussion by danthegoodman1** [duckdb/duckdb#11990] [Ideas]
I would be useful to be able to have functions to operate on the metadata of S3 files (HEAD requests). For example being able to aggregate the file sizes of an S3 glob pattern.

This is probably possible (probably omits parquet overhead) with parquet using the `total_compressed_size` function, but limited for other file formats.

Fwiw you can get the file name now with the `filename = true` option:
```sql
SELECT *
FROM read_parquet('s3://bucket/*.parquet', filename = true);
```

**Comment by Mause** [comment]
Are you looking for the results of `read_blob`/`read_text` maybe? They return the file contents, but also the stat data

**Reply by danthegoodman1** [reply]
Wouldn't that actually read the whole file though? That would be really wasteful

**Reply by danthegoodman1** [reply]
Or is there an optimization for if only `size` is selected?

**Reply by danthegoodman1** [reply]
LMAO yes https://duckdb.org/docs/guides/file_formats/read_file.html#support-for-projection-pushdown

**Reply by carlopi** [reply]
It's not super clear in the docs that no content implies only metadata is fetched, if you have ideas on how to make it more clear help is welcome!

**Reply by danthegoodman1** [reply]
I spent a while looking and searching. Maybe this could be referenced in the httpfs and s3 pages?

**Comment by danthegoodman1** [comment]
Closing as can extract from existing functions
